### ISSUE TITLE
Question/Enhancement: Native Apple Silicon builds for macOS?

### ISSUE THREAD

**Issue by AbnormalPoof** [FunkinCrew/Funkin#2985]
## What is your suggestion, and why should it be implemented?

With the recent changes to lime (see the 8.2.0-Dev branch and [#1640](https://github.com/openfl/lime/issues/1640)), building lime NDLL's and compiling projects on native ARM is now possible. 

By porting over some changes from the FunkinCrew fork of lime, I managed to hack together a build for ARM!
So I decided to compare the 2 (Native ARM vs Rosetta Intel), both of them are running off of the `main` branch.

Here are the compilation times, I am using a modified version of lime which is based of the 8.2.0-Dev branch, with changes ported from the FunkinCrew fork in order to make Funkin' compile properly.
- ARM (959.05 Seconds):
![Screenshot 2024-07-10 at 4 26 21 PM](https://github.com/FunkinCrew/Funkin/assets/86753001/735826c3-8454-4a31-bd66-7f00dc239301)

- Intel (3817.11 Seconds):
![Screenshot 2024-07-10 at 4 02 06 PM](https://github.com/FunkinCrew/Funkin/assets/86753001/a07a2c86-5708-45ca-b8b9-5d98f8592a97)

Unfortunately, aside from compilation, there kind of(?) actually isn't much to talk about in terms of performance. Both of them seem to mostly run in the same FPS. With the small differences in FPS being confined to stuff like the Freeplay Menu.

One (significant?) performance boost I did encounter was with the chart editor... mainly from spamming notes in "Live Input Mode"

- ARM

https://github.com/FunkinCrew/Funkin/assets/86753001/7f91147c-94b9-40a8-bede-1f025239a11a

- Intel

https://github.com/FunkinCrew/Funkin/assets/86753001/a5e62bd5-a450-4b42-8d2f-622854d56509

**Comment by ninjamuffin99** [comment]
Definitely something on the roadmap, as I use arm64 mac for a lot of my development!
I've been keeping an eye on progress of the PRs being made, I just haven't tried out with installation too thoroughly since I didn't want to bug up my haxe installation on my mac bwah!

**Comment by AbnormalPoof** [comment]
Managed to create a universal build by using `lipo` to combine the Intel+ARM lime NDLL's and Funkin binaries together!

![Screenshot 2024-07-12 at 8 28 50 AM](https://github.com/user-attachments/assets/24d56d9e-fc85-4fda-9602-b05ea1495c15)

**Comment by JonnycatMeow** [comment]
hell yah

**Comment by JonnycatMeow** [comment]
but how do i build it for arm64 on a intel computer

**Comment by AbnormalPoof** [comment]
I only made this script with ARM64 Macs in mind. Lime allows you to cross-compile to arm64 using the `-arm64` flag. So you can probably start there.
~~There's pretty much no point in doing this though since the Mac build is already in Intel.~~

**Comment by JonnycatMeow** [comment]
i know but i wanna do it for other mods

**Comment by ninjamuffin99** [comment]
I've been able to compile for arm64 now :) I have a branch on our internal repo that I've been getting fixed up and all that. I'll have to look into all that `lipo` stuff to make a universal binary, but hopefully next FNF update should be able to run natively on arm64 :)

**Comment by AbnormalPoof** [comment]
Sweet! Glad to see arm64 compilation went smoothly.

**Comment by AbnormalPoof** [comment]
Just did a test build and ARM64 seems to compile smoothly now as of v0.5.0!
![Screenshot 2024-09-26 at 10 57 58 AM](https://github.com/user-attachments/assets/5d958a63-3820-41df-8ed1-e2783327e0ef)
I'm not sure if I should close this PR *now* or wait until an official Apple Silicon build is out, though...

**Comment by ninjamuffin99** [comment]
yeah i actually develop and test FNF on my apple silicon laptop probably more than I do at my windows machine by now haha!

i actually completely forgot to update the apple silicon for the update though... lol.. i was getting trolled with lipo lol... I think i will close this issue however since its like 99% done and just needs to compile / publish a universal version and figure out that workflow!

**Comment by AbnormalPoof** [comment]
bless
![Screenshot 2024-09-30 at 1 11 46 PM](https://github.com/user-attachments/assets/1d3139ad-5d80-48b9-9f68-fec36d85f9e9)

**Comment by AbnormalPoof** [comment]
Interesting.
